SQL_SELECT_Tariel_Kapanadze_HW.sql

```diff
@@ -0,0 +1,215 @@
+/*
+* 1. All animation movies released between 2017 and 2019 with rate more than 1, alphabetical.
+* I didn't join category because I checked to see which category_id Animation had. other solution still would include join to link table and then writine c.name = 'Animation'
+*/
+
+SELECT 
+    f.title, 
+    f.release_year, 
+    f.rating
+FROM 
+    film f 
+LEFT JOIN 
+    film_category AS fc ON fc.film_id = f.film_id
+WHERE 
+    fc.category_id = 2 
+    AND f.release_year BETWEEN 2017 AND 2019
+    AND f.rating != 'G'
+ORDER BY 
+    f.title;
+
+
+/*
+ * 2. The revenue earned by each rental store after March 2017 (columns: address and address2 – as one column, revenue)
+ * At first I focused to find store and link it with the payment to find the revenue. since adress is only join to store table,
+ * I made a subquery for store and their revenue filtered with date and added adress later.
+ */
+
+SELECT 
+    CONCAT(a.address, ' ', COALESCE(a.address2, '')) AS full_address, 
+    store_revenue.revenue
+FROM 
+    store s
+JOIN 
+    address a ON s.address_id = a.address_id
+JOIN (
+    SELECT 
+        i.store_id, 
+        SUM(p.amount) AS revenue
+    FROM 
+        inventory i
+    JOIN 
+        rental r ON i.inventory_id = r.inventory_id
+    JOIN 
+        payment p ON r.rental_id = p.rental_id
+    WHERE 
+        p.payment_date > '2017-03-31'
+    GROUP BY 
+        i.store_id
+) AS store_revenue ON s.store_id = store_revenue.store_id
+ORDER BY 
+    store_revenue.revenue DESC;
+
+/* 3. Top-5 actors by number of movies (released after 2015) they took part in (columns: first_name, last_name, number_of_movies, sorted by number_of_movies in descending order)
+ * 
+ */
+SELECT 
+    first_name, 
+    last_name, 
+    number_of_movies
+FROM 
+    actor a 
+JOIN (
```

Comment:
You need to understand when need to be used cte/sub-queries. In this case much better to implement simple JOIN.